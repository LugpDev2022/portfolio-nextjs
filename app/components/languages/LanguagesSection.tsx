import LanguageBar from './LanguageBar';
import Subtitle from '@/app/components/Subtitle';
import SectionContainer from '../SectionContainer';
import { languagesArray } from './languagesArray';

const LanguagesSection = () => (
  <SectionContainer className='bg-[#0F2434]'>
    <Subtitle>Languages</Subtitle>
    <div className='mx-auto mt-6 max-w-xl'>
      {languagesArray.map(({ language, percentage }) => (
        <LanguageBar
          key={language}
          language={language}
          percentage={percentage}
        />
      ))}
    </div>
    <span id='projects'></span>
  </SectionContainer>
);

export default LanguagesSection;
