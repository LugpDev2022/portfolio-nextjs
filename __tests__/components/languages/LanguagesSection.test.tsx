import LanguagesSection from '@/app/components/languages/LanguagesSection';
import { render, screen } from '@testing-library/react';

describe('Tests on <LanguageSection />', () => {
  test('should match snapshot', () => {
    const { container } = render(<LanguagesSection />);

    expect(container).toMatchSnapshot();
  });

  test('should render the subtitle', () => {
    render(<LanguagesSection />);

    expect(screen.getByText('Languages')).toBeTruthy();
  });
});
