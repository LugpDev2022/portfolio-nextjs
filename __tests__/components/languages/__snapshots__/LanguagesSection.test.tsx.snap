// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <LanguageSection /> should match snapshot 1`] = `
<div>
  <section
    class="sm:py-15 px-10 py-10 bg-[#0F2434]"
  >
    <h3
      class="text-center text-2xl font-bold tracking-wide text-white"
    >
      Languages
    </h3>
    <div
      class="mx-auto mt-6 max-w-xl"
    >
      <div
        class="my-4 block w-full text-white sm:my-6 sm:flex"
      >
        <span
          class="me-5 block basis-[80px] text-left text-lg font-semibold tracking-wide sm:text-right"
        >
          Spanish
        </span>
        <div
          class="h-[25px] w-full sm:flex sm:w-10/12"
        >
          <div
            class="me-2 h-[25px] bg-[#255880]"
            data-testid="percentageBar"
            style="width: 100%;"
          />
          <span
            class="hidden text-lg sm:inline"
          >
            Native
          </span>
        </div>
      </div>
      <div
        class="my-4 block w-full text-white sm:my-6 sm:flex"
      >
        <span
          class="me-5 block basis-[80px] text-left text-lg font-semibold tracking-wide sm:text-right"
        >
          English
        </span>
        <div
          class="h-[25px] w-full sm:flex sm:w-10/12"
        >
          <div
            class="me-2 h-[25px] bg-[#255880]"
            data-testid="percentageBar"
            style="width: 75%;"
          />
          <span
            class="hidden text-lg sm:inline"
          >
            Advanced
          </span>
        </div>
      </div>
      <div
        class="my-4 block w-full text-white sm:my-6 sm:flex"
      >
        <span
          class="me-5 block basis-[80px] text-left text-lg font-semibold tracking-wide sm:text-right"
        >
          French
        </span>
        <div
          class="h-[25px] w-full sm:flex sm:w-10/12"
        >
          <div
            class="me-2 h-[25px] bg-[#255880]"
            data-testid="percentageBar"
            style="width: 17%;"
          />
          <span
            class="hidden text-lg sm:inline"
          >
            Beginner
          </span>
        </div>
      </div>
    </div>
    <span
      id="projects"
    />
  </section>
</div>
`;
